pandas/tests/reshape/test_pivot.py

```diff
@@ -1223,11 +1223,10 @@ def test_crosstab_no_overlap(self):
 
         s1 = pd.Series([1, 2, 3], index=[1, 2, 3])
         s2 = pd.Series([4, 5, 6], index=[4, 5, 6])
-
-        actual = crosstab(s1, s2)
-        expected = pd.DataFrame()
-
-        tm.assert_frame_equal(actual, expected)
+        # GH16823
+        # Setting a column with a scalar and no index should raise
```

Comment:
somethings wrong here, this should not hit this path.

Reply:
In case you can give me some more pointers, here's the pytest failure showing the path that it takes to the error
```
def test_crosstab_no_overlap(self):
        # GS 10291
    
        s1 = pd.Series([1, 2, 3], index=[1, 2, 3])
        s2 = pd.Series([4, 5, 6], index=[4, 5, 6])
    
>       actual = crosstab(s1, s2)

pandas/tests/reshape/test_pivot.py:1227: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pandas/core/reshape/pivot.py:458: in crosstab
    df['__dummy__'] = 0
pandas/core/frame.py:2459: in __setitem__
    self._set_item(key, value)
pandas/core/frame.py:2529: in _set_item
    self._ensure_valid_index(value)```

Reply:
I think it's because the intersection on columns and rows of both series is producing an empty dataframe (thus 0-len index) and trying to set `df['__dummy__'] = 0` is the line at fault. Should the crosstab fn with two non-intersecting series work correctly, or should we catch that exception before?
If it is possible, how should the output look like? Maybe we can change the behavior so that it never has to work with an empty dataframe at all?

Reply:
just I don't think you need to set the ``__dumm__`` unless ``len(df)``.